graphyte.py

```diff
@@ -64,31 +64,41 @@ def stop(self):
             self._thread.join()
             self.interval = None
 
-    def build_message(self, metric, value, timestamp):
+    def build_message(self, metric, value, timestamp, tags):
         """Build a Graphite message to send and return it as a byte string."""
         if not metric or metric.split(None, 1)[0] != metric:
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
-                    type(value).__name__))
-
-        message = u'{}{} {} {}\n'.format(
+                type(value).__name__))
+        if not (isinstance(tags, list) and all(isinstance(x, dict) for x in tags)):
+            raise TypeError('"tags" must be a list of dicts')
+        if not (all(all(isinstance(key, str) for key in tag.keys()) for tag in tags)):
+            raise TypeError('tag key must be an str')
+
+        tags_suffix = ''
+        for tag in tags:
+            for key in tag.keys():
+              tags_suffix += ';{}={}'.format(key, tag[key])
```

Comment:
It's good practice to use `str.join` rather than repeated appending -- strings are immutable in Python, so this usually recreates a new string object every time around the loop causing O(N^2) behaviour. In this case it probably doesn't matter (small number of tags), but I'd prefer good practice. So, assuming we change tags to a dict, this would become:

    tags_suffix = ''.join(';{}={}'.format(k, v) for k, v in tags.items())